Build job sometimes fail because paramiko.py3compat is missing

**What happened:**
-  job failed because `paramiko.py3compat` can not be found
- occurs sometimes, retrying usually fixes this issue.
```
+ sudo podman run --name b0c843e5-fa26-4522-a1d3-7d44f0264eb2 --cap-add sys_admin --cap-add mknod --cap-add audit_write --cap-add net_raw --security-opt apparmor=unconfined --rm -v /tmp/tmp.xP6M8V3Okc/gardenlinux/gardenlinux:/gardenlinux -v /tmp/tmp.5PUW7QARdk:/config gardenlinux/base-test:1053.0 pytest --iaas=chroot --configfile=/config/config.yaml
ImportError while loading conftest '/gardenlinux/tests/conftest.py'.
conftest.py:13: in <module>
    from helper.sshclient import RemoteClient
helper/sshclient.py:12: in <module>
    from paramiko.py3compat import u
E   ModuleNotFoundError: No module named 'paramiko.py3compat'
make: *** [Makefile:62: ali-gardener_prod] Error 4
Error: Process completed with exit code 2.
```
From nightly 17.02.2022: https://github.com/gardenlinux/gardenlinux/actions/runs/4201191350/jobs/7287962333#step:9:6688

--- a/tests/helper/sshclient.py
+++ b/tests/helper/sshclient.py
@@ -9,10 +9,19 @@ from binascii import hexlify
 from paramiko import SSHClient, AutoAddPolicy, RSAKey
 from paramiko.auth_handler import AuthenticationException, SSHException
 from paramiko.ssh_exception import NoValidConnectionsError
-from paramiko.py3compat import u
 from scp import SCPClient, SCPException
 
 logger = logging.getLogger(__name__)
+
+# paramiko.py3compat was removed with paramiko 3.0.0
+# https://www.paramiko.org/changelog.html
+def decode_unicode_or_bytes(s, encoding='utf8'):
+    """ decodes bytes or unicode to unicode"""
+    if isinstance(s, str):
+        return s
+    elif isinstance(s, bytes):
+        return s.decode(encoding)
+    raise TypeError("{!r} is not unicode or byte".format(s))
 
 
 class RemoteClient:
@@ -49,7 +58,7 @@ class RemoteClient:
                 f.write(f"{pub.get_name()} {pub.get_base64()}")
                 if comment:
                     f.write(f" {comment}")
-        hash = u(hexlify(pub.get_fingerprint()))
+        hash = decode_unicode_or_bytes(hexlify(pub.get_fingerprint()))
         fingerprint = ":".join([hash[i : 2 + i] for i in range(0, len(hash), 2)])
         logger.info(f"fingerprint: {bits} {fingerprint} {filename}.pub (RSA)")
         return fingerprint

